runtime/gc_realtime/RealtimeAccessBarrier.cpp

```diff
@@ -500,27 +482,10 @@ MM_RealtimeAccessBarrier::jniGetStringCritical(J9VMThread* vmThread, jstring str
 	bool shouldCopy = false;
 	bool hasVMAccess = false;
 
-#if !defined(J9VM_GC_ARRAYLETS)
-	if ((javaVM->runtimeFlags & J9_RUNTIME_ALWAYS_COPY_JNI_CRITICAL) == J9_RUNTIME_ALWAYS_COPY_JNI_CRITICAL) {
-		VM_VMAccess::inlineEnterVMFromJNI(vmThread);
-		hasVMAccess = true;
-		shouldCopy = true;
-	} else if (IS_STRING_COMPRESSION_ENABLED_VM(javaVM)) {
-		/* If the string bytes are in compressed UNICODE, then we need to copy to decompress */
-		VM_VMAccess::inlineEnterVMFromJNI(vmThread);
-		hasVMAccess = true;
-		J9Object *stringObject = (J9Object*)J9_JNI_UNWRAP_REFERENCE(str);
-		if (IS_STRING_COMPRESSED(javaVM, stringObject)) {
-			isCompressed = true;
-			shouldCopy = true;
-		}
-	}
-#else /* J9VM_GC_ARRAYLETS */
-	// For now only copying is supported for arraylets
+	/* For now only copying is supported for arraylets */
 	VM_VMAccess::inlineEnterVMFromJNI(vmThread);
 	hasVMAccess = true;
 	shouldCopy = true;
```

Comment:
I believe there is no need to initialize these variables to `false` earlier. We can declare variables here